### ISSUE TITLE
Confusion Matrices YOLOv8 OBB

### ISSUE THREAD

**Issue by mmatteo-hub** [ultralytics/ultralytics#8356]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Train, Val, Predict

### Bug

# Oriented Bounding Boxes

### Plots
I am facing a problem in displaying the confusion matrices after the train process.
<img src="https://github.com/ultralytics/ultralytics/assets/62358773/47583d86-080b-4c1e-94b8-86a3f005ac16" width=25%, height=25%> <img src="https://github.com/ultralytics/ultralytics/assets/62358773/d31928c0-a12b-4919-a7d9-8057256d35b9" width=25%, height=25%>

The labels and correlogram plot are shown correctly.
<img src="https://github.com/ultralytics/ultralytics/assets/62358773/f792b89f-e077-4b57-ab2f-c2112ed60c5f" width=25%, height=25%> <img src="https://github.com/ultralytics/ultralytics/assets/62358773/966c21f3-d002-4b20-8778-5ce86c3b254e" width=25%, height=25%>

### `YAML` File
The `.yaml` used is in this format
```YAML
path: path_to_folder
train: train\images
val: val\images

# Classes
names:
  0: 'class1'
```

Also tried in this format
```YAML
train: path_to_folder\train
val: path_to_folder\val

nc: 1

names: ['class1']
```

## Folder Structure
folder\
│
├── train\
│   ├── images\
│   └── labels\
│
└── val\
    ├── images\
    └── labels\

## Models Used
These are the models testes:
* `yolov8x-obb.pt`
* `yolov8n-obb.pt`

### Environment

Ultralytics YOLOv8.1.4 🚀 Python-3.11.7 torch-2.1.2+cpu CPU (11th Gen Intel Core(TM) i7-1165G7 2.80GHz)
Setup complete ✅ (8 CPUs, 15.7 GB RAM, 219.6/465.4 GB disk)

OS                  Windows-10-10.0.22621-SP0
Environment         Windows
Python              3.11.7
Install             pip
RAM                 15.74 GB
CPU                 11th Gen Intel Core(TM) i7-1165G7 2.80GHz
CUDA                None

matplotlib          ✅ 3.8.0>=3.3.0
numpy               ✅ 1.26.3>=1.22.2
opencv-python       ✅ 4.9.0.80>=4.6.0
pillow              ✅ 10.0.1>=7.1.2
pyyaml              ✅ 6.0.1>=5.3.1
requests            ✅ 2.31.0>=2.23.0
scipy               ✅ 1.12.0>=1.4.1
torch               ✅ 2.1.2>=1.8.0
torchvision         ✅ 0.16.2>=0.9.0
tqdm                ✅ 4.66.1>=4.64.0
psutil              ✅ 5.9.0
py-cpuinfo          ✅ 9.0.0
thop                ✅ 0.1.1-2209072238>=0.1.1
pandas              ✅ 2.2.0>=1.1.4
seaborn             ✅ 0.13.1>=0.11.0

## Code
This is the Python script used to 

```Python
from ultralytics import YOLO
import os

os.environ['KMP_DUPLICATE_LIB_OK']='True'

model = YOLO('yolov8x-obb.pt')

data = 'path_to_.yaml'

# Not relevant to reproduce the bug since it is not related to these parameters
epochs = ...
imgsz = ...
batch = ...
patience = ...

results = model.train(data=data, epochs=epochs, imgsz=imgsz, batch=batch, patience=patience)
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@mmatteo-hub OBB confusion matrix is working for me. If nothing is showing up you may want to check your validation data to make sure that your paths are correct, and also train longer as there may not be any detections above threshold due to a poorly trained model.

<img width="1528" alt="Screenshot 2024-02-21 at 11 15 14" src="https://github.com/ultralytics/ultralytics/assets/26833433/cb8e962c-b68c-4bda-aa6f-7411f68bf387">

**Comment by KybernetikJo** [comment]
@mmatteo-hub 
If your information is correct, you use version 8.1.4!
Have you tried version 8.1.16? #8299

**Comment by mmatteo-hub** [comment]
@glenn-jocher I tried upgrading the version using pip and now it has started working again. Hope also in the future. Thank you

**Comment by glenn-jocher** [comment]
@mmatteo-hub that's great to hear! 😊 If you encounter any more issues or have questions in the future, feel free to reach out. Happy coding!
